src/Native/SymSgdNative/CMakeLists.txt

```diff
@@ -1,25 +1,46 @@
 project (SymSgdNative)
 
+
 set(SOURCES
     SymSgdNative.cpp
 )
 
-find_library(MKL_LIBRARY MklImports HINTS ${MKL_LIB_PATH})
-if(NOT WIN32)
+if(APPLE)
+    # CMake has support for OpenMP, however, Apple has a version of Clang
+    # that does not support openMP out of the box. Therefore
+    # these commands are added to sepcifically handle the Apple Clang scenario
+    # If the LLVM version of clang is used for Apple builds, this can be removed
+    # and the else condition can be used instead.
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xpreprocessor -fopenmp")
+    SET(OPENMP_LIBRARY "omp")
+    include_directories("/usr/local/opt/libomp/include")
+    link_directories("/usr/local/opt/libomp/lib")
+
     list(APPEND SOURCES ${VERSION_FILE_PATH})
-    if(NOT APPLE)
+else()
+    find_package(OpenMP)
+    if (OPENMP_FOUND)
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+        SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+    endif()
+
+    if (NOT WIN32)
```

Comment:
This is not correct. This check will never happen on an APPLE system.

Please fix this in a new PR.

Reply:
Ah, I see it now. It is in both conditions.

We could have just made that a completely separate `if`... but I guess this is OK. It is just generally "odd" to have APPLE different and Windows/LINUX in the same condition.